### PR TITLE
Hotel 14

### PR DESCRIPTION
**Pull request by dudarenko-dmitry**

### CODE REVIEW

File: core/src/main/java/pl/senla/hotel/comparators/HotelServicesComparatorByPrice.java [hunk 1]
```diff
@@ -1,25 +1,26 @@
...
 import java.util.Comparator;
 import java.util.Objects;
+import java.util.Optional;
 
 import static pl.senla.hotel.constant.HotelServiceConstant.ERROR_IN_SERVICE_TYPE;
 
-@AppComponent
+@Component
 @Slf4j
 public class HotelServicesComparatorByPrice implements Comparator<HotelService> {
```
Comment (Mark005) [inline core/src/main/java/pl/senla/hotel/comparators/HotelServicesComparatorByPrice.java]: why do you still need this? sort it in db

File: core/src/main/java/pl/senla/hotel/connection/AbstractConnection.java [hunk 2]
```diff
@@ -1,37 +1,37 @@
...
-                log.debug(DB_CONNECTED);
-            }
-            return connection;
-        } catch (SQLException e) {
-            log.error(DB_CONNECTION_ERROR);
-            throw new SQLException(DB_CONNECTION_ERROR, e);
-        }
-    }
-}
+//package pl.senla.hotel.connection;
```
Comment (Mark005) [inline core/src/main/java/pl/senla/hotel/connection/AbstractConnection.java]: remove if it's not used

File: core/src/main/java/pl/senla/hotel/entity/services/HotelService.java [hunk 3]
```diff
@@ -51,16 +50,16 @@ public class HotelService implements Serializable {
...
-    @Column(name = "checkOut", nullable = false)
+    @Column(name = "checkOut")
     private LocalDateTime checkOutTime;
 
     @Column(name = "cost", nullable = false)
     private Integer cost;
 
-    @GetInstance(beanName = "ServiceFacilityDB")
+    @Autowired
     private transient ServiceFacility serviceRoom;
```
Comment (Mark005) [inline core/src/main/java/pl/senla/hotel/entity/services/HotelService.java]: what is this? why it's here? remove

File: core/src/main/java/pl/senla/hotel/service/ServiceOrderSpring.java [hunk 4]
```diff
@@ -1,105 +1,109 @@
...
     }
 
     @Override
-    public boolean update(int idOrder, String orderUpdatingString) {
+    public Order update(int idOrder, String orderUpdatingString) {
 //        if (read(idOrder) != null) {
 //            Order orderUpdate = new Order(); // read from String
 //            return daoOrder.update(idOrder, orderUpdate);
 //        }
         log.debug(ORDER_NOT_EXISTS);
```
Comment (Mark005) [inline core/src/main/java/pl/senla/hotel/service/ServiceOrderSpring.java]: where is your implementation?

File: core/src/main/java/pl/senla/hotel/utils/DBPropertiesUtil.java [hunk 5]
```diff
@@ -1,46 +1,46 @@
...
+//            throw new RuntimeException("System hasn't read property-file", e);
+//        }
+//    }
+//
+//    private Properties loadProperty(String configDirectory, String configName) {
+//        Properties properties = new Properties();
+//        try (FileInputStream fis = new FileInputStream(new File(configDirectory, configName))) {
+//            properties.load(fis);
+//        } catch (IOException e) {
+//            throw new RuntimeException(e);
```
Comment (Mark005) [inline core/src/main/java/pl/senla/hotel/utils/DBPropertiesUtil.java]: remove

File: ui/src/main/java/pl/senla/hotel/Main.java [hunk 6]
```diff
@@ -1,21 +1,34 @@
...
-        log.debug(START_APPLICATION);
-        DIApplication.getApplication().run();
+        log.info(START_APPLICATION);
+
+        // previous version 1
+//        DIApplication.getApplication().run();
+
+//        version 3:
+        ApplicationContext ctx = new AnnotationConfigApplicationContext(SpringConfig.class);
+        StartMenuMainDB startMenuMainDB = ctx.getBean(StartMenuMainDB.class);
```
Comment (Mark005) [inline ui/src/main/java/pl/senla/hotel/Main.java]: please read about post construct annotation and move it into StartMenuMainDB class. It's a very bad practice to use app context in your business logic there is always a way to solve it in a 'right' way

File: ui/src/main/java/pl/senla/hotel/SpringConfig.java [hunk 7]
```diff
@@ -0,0 +1,14 @@
+package pl.senla.hotel;
+
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
+
+@Configuration
+@ComponentScan("pl.senla.hotel")
+@PropertySource("application.properties")
```
Comment (Mark005) [inline ui/src/main/java/pl/senla/hotel/SpringConfig.java]: @ComponentScan
@PropertySource
@EnableJpaRepositories
are not required if you are using Spring Boot please read what goes with Spring Boot out of the box and also you can read what @SpringBootApplication hides inside

File: ui/src/main/java/pl/senla/hotel/ui/analytic/ExecutorAnalyticsDB.java [hunk 8]
```diff
@@ -1,27 +1,32 @@
...
 import pl.senla.hotel.ui.Executor;
 
 import java.lang.reflect.InvocationTargetException;
 import java.util.Scanner;
 
 import static pl.senla.hotel.constant.ConsoleConstant.*;
 
-@AppComponent
+@Component
+@Qualifier("ExecutorAnalyticsDB")
```
Comment (Mark005) [inline ui/src/main/java/pl/senla/hotel/ui/analytic/ExecutorAnalyticsDB.java]: please read about Qualifier annotation and how it should be used

File: ui/src/main/java/pl/senla/hotel/ui/analytic/NavigatorAnalytics.java [hunk 9]
```diff
@@ -1,20 +1,21 @@
...
+import org.springframework.stereotype.Component;
 import pl.senla.hotel.ui.Item0QuitToMain;
 import pl.senla.hotel.ui.MenuItem;
 import pl.senla.hotel.ui.Navigator;
 
 import static pl.senla.hotel.constant.MenuConstant.MENU_ANALYTICS_REPORTS;
 
-@AppComponent
+@Component
+@Qualifier("NavigatorAnalytics")
```
Comment (Mark005) [inline ui/src/main/java/pl/senla/hotel/ui/analytic/NavigatorAnalytics.java]: same here

**Review by Mark005 — CHANGES_REQUESTED** [review]

File: core/src/main/java/pl/senla/hotel/connection/AbstractConnection.java (hunk 2 above)
Comment (dudarenko-dmitry) [reply core/src/main/java/pl/senla/hotel/connection/AbstractConnection.java]: ready

File: ui/src/main/java/pl/senla/hotel/SpringConfig.java (hunk 7 above)
Comment (dudarenko-dmitry) [reply ui/src/main/java/pl/senla/hotel/SpringConfig.java]: ready

File: ui/src/main/java/pl/senla/hotel/Main.java (hunk 6 above)
Comment (dudarenko-dmitry) [reply ui/src/main/java/pl/senla/hotel/Main.java]: ready

File: core/src/main/java/pl/senla/hotel/utils/DBPropertiesUtil.java (hunk 5 above)
Comment (dudarenko-dmitry) [reply core/src/main/java/pl/senla/hotel/utils/DBPropertiesUtil.java]: ready

File: core/src/main/java/pl/senla/hotel/entity/services/HotelService.java (hunk 3 above)
Comment (dudarenko-dmitry) [reply core/src/main/java/pl/senla/hotel/entity/services/HotelService.java]: ready

File: ui/src/main/java/pl/senla/hotel/ui/analytic/ExecutorAnalyticsDB.java (hunk 8 above)
Comment (dudarenko-dmitry) [reply ui/src/main/java/pl/senla/hotel/ui/analytic/ExecutorAnalyticsDB.java]: ready

File: ui/src/main/java/pl/senla/hotel/ui/analytic/NavigatorAnalytics.java (hunk 9 above)
Comment (dudarenko-dmitry) [reply ui/src/main/java/pl/senla/hotel/ui/analytic/NavigatorAnalytics.java]: ready

File: core/src/main/java/pl/senla/hotel/comparators/HotelServicesComparatorByPrice.java (hunk 1 above)
Comment (dudarenko-dmitry) [reply core/src/main/java/pl/senla/hotel/comparators/HotelServicesComparatorByPrice.java]: ready

**Review by Mark005 — APPROVED** [review]
now It looks much better, great job!

**Pull request merged**